apps/storefront/src/components/layout/B3Nav.tsx

```diff
@@ -1,30 +1,61 @@
-import { useContext } from 'react';
-import { useLocation, useNavigate } from 'react-router-dom';
+import { useContext, useEffect, useMemo } from 'react';
+import { matchPath, useLocation, useNavigate } from 'react-router-dom';
 import { useB3Lang } from '@b3/lang';
 import { Badge, List, ListItem, ListItemButton, ListItemText, useTheme } from '@mui/material';
 
 import { useMobile } from '@/hooks';
-import { DynamicallyVariableedContext } from '@/shared/dynamicallyVariable';
+import { DynamicallyVariedContext } from '@/shared/dynamicallyVariable';
 import { GlobalContext } from '@/shared/global';
-import { getAllowedRoutes } from '@/shared/routes';
-import { useAppSelector } from '@/store';
+import { getAllowedRoutes, RouteItem } from '@/shared/routes';
+import { store, useAppSelector } from '@/store';
+import {
+  setCompanyHierarchyInfoModules,
+  setPagesSubsidiariesPermission,
+} from '@/store/slices/company';
+import { PagesSubsidiariesPermissionProps } from '@/types';
 import { B3SStorage } from '@/utils';
+import { verifyCompanyLevelPermissionByCode } from '@/utils/b3CheckPermissions';
 
 import { b3HexToRgb, getContrastColor } from '../outSideComponents/utils/b3CustomStyles';
 
 interface B3NavProps {
   closeSidebar?: (x: boolean) => void;
 }
 
+const getSubsidiariesPermission = (routes: RouteItem[]) => {
+  const subsidiariesPermission = routes.reduce((all, cur) => {
+    if (cur?.subsidiariesCompanyKey) {
+      const code = cur.permissionCodes?.includes(',')
+        ? cur.permissionCodes.split(',')[0].trim()
+        : cur.permissionCodes;
```

Comment:
Why is this needed? Seems like data clean up that should be done by the API.

Reply:
Thank you for pointing this out!

This logic is not about "data cleaning" but is instead tied directly to our business rules for permission handling.    In our b2bPermissionsList, permission codes are often grouped as comma-separated strings to represent multiple related permissions (e.g., 'get_users, get_user_detail').

The first code in each string serves as the primary identifier for permission checks.    Extracting the first code ensures consistent and efficient validation of user permissions without requiring downstream components to handle string splitting repeatedly.

For example, a "user management" View permission corresponds to 'get_users, get_user_detail'. I only need to extract one of these codes to compare it against the permissions retrieved at login, ensuring the required permission is present.